include/spdlog/sinks/eventlog_sink_win32.h

```diff
@@ -0,0 +1,246 @@
+        if (msg.level >= level::debug)
+        {
+            category = 2;
+            if (msg.level >= level::info)
+            {
+                category = 3;
+                if (msg.level >= level::warn)
+                {
+                    category = 4;
+                    if (msg.level >= level::err)
+                    {
+                        category = 5;
+                        if (msg.level >= level::critical)
+                        {
+                            category = 6;
+                        }
+                    }
+                }
+            }
+        }
+        return category;
+    }
+};
+
+class sink : public base_sink<spdlog::details::null_mutex>
+{
+private:
+    HANDLE hEventLog_;
+    SID* current_user_sid_;
+    std::string source_;
+    std::string log_;
+
+    void add_registry_info(std::string const& message_file_path)
+    {
+        std::string subkey("SYSTEM\\CurrentControlSet\\Services\\EventLog\\");
+        subkey += log_ + "\\" + source_;
+
+        ::HKEY hkey {};
+        DWORD disposition {};
+        long stat = RegCreateKeyEx(HKEY_LOCAL_MACHINE, subkey.c_str(), 0, NULL,
+                                   REG_OPTION_NON_VOLATILE, KEY_SET_VALUE, NULL,
+                                   &hkey, &disposition);
+        if (stat == ERROR_SUCCESS)
+        {
+            if (disposition == REG_CREATED_NEW_KEY && !message_file_path.empty())
+            {
+                auto const expanded_message_file_path_length = ExpandEnvironmentStrings(message_file_path.c_str(), (LPSTR) &disposition, 0);
+                std::vector<char>expanded_message_file_path(expanded_message_file_path_length + 1);
+                ExpandEnvironmentStrings(message_file_path.c_str(), expanded_message_file_path.data(), expanded_message_file_path_length);
+
+                RegSetValueEx(hkey, "EventMessageFile", 0, REG_SZ, (LPBYTE) expanded_message_file_path.data(), expanded_message_file_path_length);
+                DWORD typesSupported = 7;
+                RegSetValueEx(hkey, "TypesSupported", 0, REG_DWORD, (LPBYTE) &typesSupported, sizeof(DWORD));
+            }
+        }
+        else
+        {
+            SPDLOG_THROW(win32_error("RegCreateKeyEx", stat));
+        }
+
+        RegCloseKey(hkey);
+    }
+
+protected:
+    void sink_it_(const details::log_msg &msg) override
+    {
+        memory_buf_t formatted;
+        formatter_->format(msg, formatted);
+
+        auto formatted_string = fmt::to_string(formatted);
+        auto formatted_string_lpsz = formatted_string.c_str();
+
+        if (!ReportEvent(hEventLog_, eventlog::get_event_type(msg), eventlog::get_event_category(msg), 1000, current_user_sid_, 1, 0, &formatted_string_lpsz, nullptr))
+            SPDLOG_THROW(win32_error("ReportEvent"));
+    }
+
+    void flush_() override {}
+
+public:
+    sink(
+        std::string const& source, 
+        std::string const& log, 
+        std::string const& message_file_path) 
```

Comment:
Is this "message_file_path" param really needed? Most users will just be happy with default dll